yaml/yaml.go

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+*
+* Copyright 2024 SAP SE
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You should have received a copy of the License along with this
+* program. If not, you may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*******************************************************************************/
+
+// This package is just an aid to migrate from gopkg.in/yaml.v2 to gopkg.in/yaml.v2
+// It provides a helper to parse a given yaml document with both major versions and log if there is a difference.
+package yaml
+
+import (
+	"reflect"
+
+	"github.com/google/go-cmp/cmp"
+	yaml_v2 "gopkg.in/yaml.v2"
+	yaml_v3 "gopkg.in/yaml.v3"
+
+	"github.com/sapcc/go-bits/logg"
+)
+
+func Unmarshal(in []byte, out any) error {
+	var outV3 any
```

Comment:
This will fail for any non-trivial type on `out` because `outV3` will always be filled with a `map[any]any` if the top-level structure of the YAML file is a dictionary. You need to ensure that `out` and `outV3` have the same underlying type:
```suggestion
func Unmarshal[T any](in []byte, out *T) error {
	var outV3 T
```